### PR TITLE
Add `team` to `OAuthApplicationInfo`

### PR DESCRIPTION
**Pull request by eritbh**
References: [Get OAuth Application request](https://discord.com/developers/docs/topics/oauth2#get-current-application-information) and [Team data models](https://discord.com/developers/docs/topics/teams#data-models)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -849,6 +849,19 @@ declare namespace Eris {
...
+    icon: string | null;
+    id: string;
+    members: OAuthTeamMember;
+    owner_user_id: string;
+  }
+  interface OAuthTeamMember {
+    membership_state: number;
+    permissions: string[];
+    team_id: string;
+    user: { username: string; discriminator: string; id: string; avatar: string | null };
```
Comment (bsian03) [inline index.d.ts]: Can you sort the props alphabetically as well plz

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (eritbh) [reply index.d.ts]: This order was copy pasted from elsewhere. Want me to alphabetize there too?

Comment (bsian03) [reply index.d.ts]: Yeah that would be great thanks

Comment (eritbh) [reply index.d.ts]: Broken out into a new `PartialUser` interface with alphabetized keys. Also used in several other places, correcting some other discrepancies as well (in particular, `avatar` is nullable, not optional).

File: index.d.ts [hunk 2]
```diff
@@ -2247,6 +2255,13 @@ declare namespace Eris {
     removeRelationship(): Promise<void>;
   }
 
+  export interface PartialUser {
```
Comment (bsian03) [inline index.d.ts]: Put this in the Member interface category and rename the category to Member/User

**Review by bsian03 — CHANGES_REQUESTED** [review]

**Pull request merged**